pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -1,153 +1,226 @@
 {
+  lib,
+  stdenv,
   bzip2,
   callPackage,
-  expat,
   fetchFromGitHub,
   fontconfig,
   freetype,
   glib,
   glslang,
   harfbuzz,
-  lib,
-  libadwaita,
   libGL,
-  libpng,
   libX11,
-  libXcursor,
-  libXi,
-  libXrandr,
+  libadwaita,
   ncurses,
   nixosTests,
   oniguruma,
   pandoc,
   pkg-config,
   removeReferencesTo,
-  stdenv,
   versionCheckHook,
   wrapGAppsHook4,
   zig_0_13,
-  zlib,
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/apprt.zig#L72-L76
+  appRuntime ?
+    {
+      linux = "gtk";
+      windows = "glfw";
+    }
+    .${stdenv.hostPlatform.parsed.kernel.name} or "none",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/font/main.zig#L94
+  fontBackend ? if stdenv.hostPlatform.isDarwin then "coretext" else "fontconfig_freetype",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/PACKAGING.md#build-options
+  optimizeLevel ? "ReleaseFast",
```

Comment:
Do we really need this? It doesn't sound like any user could possibly want to use anything but ReleaseFast.

Reply:
[It was asked for previously](https://github.com/NixOS/nixpkgs/pull/368404#discussion_r1898402912). I think it would also make sense to expose this directly, as the way we override `zig.hook` makes it difficult to change this flag directly

I've also found it useful in iterating on this expression at least :p